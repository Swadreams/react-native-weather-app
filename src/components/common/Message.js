import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {COLORS} from '../../constants/Colors';

const Message = ({message, isError, style}) => {
  return (
    <View style={styles.container}>
      <Text style={[isError ? styles.errorMessage : styles.message, style]}>
        {message}
      </Text>
    </View>
  );
};

const messageStyle = {
  fontSize: 18,
  alignItems: 'center',
  textAlign: 'center',
  minHeight: 46,
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: 24,
  },
  message: {
    ...messageStyle,
    color: COLORS.white,
  },
  errorMessage: {
    ...messageStyle,
    color: COLORS.red,
  },
});

export default React.memo(Message);
