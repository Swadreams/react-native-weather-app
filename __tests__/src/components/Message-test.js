import 'react-native';
import React from 'react';
import Message from '../../../src/components/common/Message';
import renderer from 'react-test-renderer';

test('Message snapshot ', () => {
  const snap = renderer.create(<Message message={'welcome'} />).toJSON();
  expect(snap).toMatchSnapshot();
});
