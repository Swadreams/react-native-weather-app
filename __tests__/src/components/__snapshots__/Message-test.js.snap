// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message snapshot  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "flex-start",
      "padding": 24,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignItems": "center",
          "color": "#fff",
          "fontSize": 18,
          "minHeight": 46,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    welcome
  </Text>
</View>
`;
